EMS/BAL/EmployeeBAL.cs

```diff
@@ -0,0 +1,158 @@
+using EmployeeManagementSystem.DAL;
+using EmployeeManagementSystem.Models;
+using EmployeeManagementSystem.Utils;
+namespace EmployeeManagementSystem.BAL;
+
+public class EmployeeBAL : IEmployeeBAL
+{
+    private readonly IEmployeeDAL _employeeDal;
+    private readonly IDropdownDAL _dropdownDal;
+    private readonly ILogger _logger;
+
+    public EmployeeBAL(ILogger logger, IEmployeeDAL employeeDal, IDropdownDAL dropdownDAL)
+    {
+        _employeeDal = employeeDal;
+        _dropdownDal = dropdownDAL;
+        _logger = logger;
+    }
+
+    public List<EmployeeDetails>? GetAll()
+    {
+        List<EmployeeDetails> employeeDetailsList;
+        try
+        {
+            employeeDetailsList = GetEmployeeDetails();
+        }
+        catch (Exception)
+        {
+            throw; 
+        }
+        return employeeDetailsList;
+    }
+
+    public bool AddEmployee(Employee employee)
+    {
+        bool status;
+        try
+        {
+            status = _employeeDal.Insert(employee);
+        }
+        catch (Exception)
+        {
+            status = false;
+        }
+        return status;
+    }
+
+    public bool DeleteEmployees(string empNo)
+    {
+        bool status;
+        try
+        {
+            status = _employeeDal.Delete(empNo);
+        }
+        catch (Exception)
+        {
+            status = false;
+        }
+
+        return status;
+    }
+
+    public bool UpdateEmployee(string empNo, Employee employee)
+    {
+        try
+        {
+            return _employeeDal.Update(empNo, employee);
+        }
+        catch (Exception ex)
+        {
+            throw new Exception("Error occurred while updating employee data.", ex);
+        }
+    }
+
+    public List<EmployeeDetails>? SearchEmployees(EmployeeFilters keyword)
+    {
+        List<EmployeeDetails> employeeDetailsList, resultEmployees;
+        try
+        {
+            employeeDetailsList =  employeeDetailsList = GetEmployeeDetails();
```

Comment:
this approach is not good. filter should happen at db layer only. and you should call only filter method.